### PR TITLE
Updated docs on the need to load confidential-identities CorDapp separately

### PR DESCRIPTION
**Pull request by shamsasari**
Also, updated the trader and BoC demos to include confidential-identities jar.

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -287,6 +287,11 @@ The TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 family of ciphers is retired from the li
 as it is a legacy cipher family not supported by all native SSL/TLS implementations. We anticipate that this
 will have no impact on any deployed configurations.
 
+Confidential identities
++++++++++++++++++++++++
+
+If any of your CorDapps use the experimental confidential-identities module then it also needs to be loaded as a separate CorDapp jar.
+
 Miscelleneous changes
```
Comment (josecoll) [inline docs/source/release-notes.rst]: "miscellaneous" (I know its not your mis-spelling)

File: docs/source/release-notes.rst [hunk 2]
```diff
@@ -287,6 +287,11 @@ The TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 family of ciphers is retired from the li
 as it is a legacy cipher family not supported by all native SSL/TLS implementations. We anticipate that this
 will have no impact on any deployed configurations.
 
+Confidential identities
++++++++++++++++++++++++
+
+If any of your CorDapps use the experimental confidential-identities module then it also needs to be loaded as a separate CorDapp jar.
```
Comment (josecoll) [inline docs/source/release-notes.rst]: I think we need to add a huge warning here that highlights that Finance directly depends on the Confidential Identities CorDapps for any anonymous identity usage in transactions.

Comment (shamsasari) [reply docs/source/release-notes.rst]: Updated to mention finance. A huge warning is too much.

**Comment by shamsasari** [comment]
Back to you

File: docs/source/release-notes.rst [hunk 3]
```diff
@@ -287,7 +287,13 @@ The TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 family of ciphers is retired from the li
 as it is a legacy cipher family not supported by all native SSL/TLS implementations. We anticipate that this
 will have no impact on any deployed configurations.
 
-Miscelleneous changes
+Confidential identities
++++++++++++++++++++++++
+
+If any of your CorDapps use the experimental confidential-identities module then it also needs to be loaded as a separate CorDapp jar. This
+includes the incubating Corda finance CorDapp.
```
Comment (josecoll) [inline docs/source/release-notes.rst]: Is it really "incubating"? 
Mike asked me specifically to rename it "Corda Finance Demo"

Comment (shamsasari) [reply docs/source/release-notes.rst]: The phrase came from here https://docs.corda.net/head/corda-api.html#corda-incubating-modules. I can change it to demo.

**Comment by shamsasari** [comment]
Back to you

**Pull request merged**